### ISSUE TITLE
Add Sign In With Apple method

### ISSUE THREAD

**Issue by raz0r007** [heroiclabs/nakama-cpp#35]
In nakama-cpp library, these are the methods that we can use in order to authenticate a player to Nakama Server:

```
virtual void authenticateDevice(.....)
virtual void authenticateEmail(.....)
virtual void authenticateFacebook(.....)
virtual void authenticateGoogle(.....)
virtual void authenticateGameCenter(.....)
virtual void authenticateCustom(.....)
virtual void authenticateSteam(.....)
```

These methods works perfect, but we **CAN'T** authenticate through Apple.
If we use a third-party login service, Apple is forcing developers to implement Sign in with Apple (GameCenter is not enough).

Nakama Server is already supporting it: https://github.com/heroiclabs/nakama/releases/v2.13.0
But we need to add this feature to nakama-cpp library.

Topic related: https://forum.heroiclabs.com/t/sign-in-with-apple/1057

Thanks

**Comment by Dimon4eg** [comment]
It's already added to nakama-cpp but not released yet.

**Comment by raz0r007** [comment]
Thanks @Dimon4eg 
Do we have any estimated date? This is very important. It's a **blocker**, because Apple doesn't want to approve the binary. It means that we can't neither publish our app nor test using TestFlight.

@novabyte mention that it's "easy" to implement. I'd like to inspect the PR if you create it, in order to learn how to contribute in this things.

Thanks

**Comment by Dimon4eg** [comment]
I will release 1-2 November

**Comment by Dimon4eg** [comment]
Released version 2.4.0
